### DISCUSSION TITLE
Offline only mode

### DISCUSSION THREAD

**Discussion by TimelessNL** [johanmeijer/grott#318] [Q&A] [answered]
Hi,

Great work on the project so far!

I've read through the repository and maybe I've missed it, but I cannot find a clear answer if `grott` supports offline only mode.
Is my assumption correct that if data is not proxied to _server.growatt.com_ the inverter does not attempt to send any data and no traffic can be intercepted? As mentioned in a 2 year old issue #11?

I also came across [Growatt-server](https://github.com/knowthelist/Growatt-server) which seems to be a perl implementation of the growatt servers, maybe this is something to be integrated in `grott` as offline mode?

**Comment by johanmeijer** [comment, marked as answer]
Yes your assumption is correct . Grott only proxied the data to server.growatt.com, but .......

we also have grottserver that act as a replacement for server.growatt.com (and can be used to send via an API interface commands to the inverter. 

Although I still state grotsserver is in beta it works really fine (I, and others, use it for ; months know). Beta is more because I have not document it very well at this moment. You can read more about it in: #98 (discussions: New feature: grottserver).

**Comment by TimelessNL** [comment]
That is indeed a really nice option to have! I think it would be great to have a short mention of it in the [data retrieval section](https://github.com/johanmeijer/grott#two-modes-of-metric-data-retrieval) even while it is in beta. That way people like me that couldn't find the proper search terms can easily find it 😄

**Comment by johanmeijer** [comment]
It is already in the read.me: 

![image](https://user-images.githubusercontent.com/16508298/229354185-376e74e1-def4-4ac7-b97d-4f062e6bd1cc.png)

But I will document it better in the near future.

**Comment by TimelessNL** [comment]
Ah indeed, sorry I missed that. 👍 

One off-topic question though. I'm still waiting for my `Growatt ShineWiFi-X` to arrive, but while I'm waiting I took the liberty to setup and configure the [grottserver docker container](https://hub.docker.com/r/ledidobe/grottserver) which went seamless. One thing I noticed is that there is no incoming connection on my `MQTT` server yet, Am I correct to assume that grott will only connect to MQTT once it has received/intercepted communication from the inverter?

**Comment by johanmeijer** [comment]
Yes Grott will only connect to the MQTT server to send a message (fire and forget). At this moment it is not able to have a persistent session with MQTT. See also issue #301.
